documentation/API_documentation/QoD_Stable_Bandwidth_API.md

```diff
@@ -0,0 +1,180 @@
+
+<img src="./resources/QoD_bandwidth_overview.PNG" alt="QoD_BM" title="QoD Bandwidth API Overview">
+
+
+## 2\. Quick Start
+
+The usage of the stable throughput API is based on QoS sessions, which can be created (based on available QoS profiles), queried and deleted.
+The deletion of a requested session can be triggered by the user or can be triggered automatically. The automatic process is triggered either when the user specified duration has reached its limit or default session expiration time has been reached (currently set to 24hrs). 
+
+Before starting with the API, the developer needs to provide base-URL, security credentials and App-Flow parameters with specified QoS profile.
+
+**Base-URL**
+The RESTful Stable Throughput API endpoint, for example [**https://telekom-api.developer.telekom.com/5g-latency**](https://telekom-api.developer.telekom.com/5g-throughput)
+
+**Authentication**
+Configure security access keys such as O-Auth2 client credentials to be used by Client applications which will invoke the QoD API.
+
+**QoS Profile**
+Define throughput requirements of the application and identify QoS profile class which maps into the required performance category.
+
+**App-Flow**
+Describes the precise flow the developer wants to prioritize and have stable bandwidth for. This flow is described using source and destination IP addresses, ports and protocols with flow direction.
+
+**Duration**
+Define the number of seconds for which the QoD session should be created. This parameter is optional and if not specified, the session is either deleted on user request or if default expiration limit of 24hrs has been reached.
+
+**Notification URL and token**
+Developers have a chance to specify callback URL on which notifications (eg. session termination) regarding the session can be received from the service provider. This is also an optional parameter.
+
+Sample API invocations are presented in Section 4.5.
+
+## 3\. Authentication and Authorization
+
+The QoD Service API makes use of the client credentials grant which is applicable for server to server use cases involving trusted partners or clients without any protected user data involved.
+In this method the API invoker client is registered as a confidential client with an authorization grant type of client\_credentials [3].
+
+## 4\. API Documentation
+
+### 4.1 Details
+The usage of the QoD throughput API is based on QoS profile classes and parameters which define App-Flows.
+Based on the API, QoS sessions can be created, queried, and deleted. Once a proper QoS profile throughput class is requested, application users get a prioritized service with stable throughput even in the case of congestion.The QoD bandwidth API has the following characteristics:
```

Comment:
I would remove the sentence 'Once a proper QoS profile throughput class is requested, application users get a prioritized service with stable throughput even in the case of congestion.' because radio offers no guarantees since there are too many variables (environment, received signal strength etc.), and this is also not a 'prioritisation' service, rather it attempts to keep throughput stable. We need to discuss what wording to replace that with in the calls.

Reply:
Agree to discuss in call.

Reply:
As of call on July 29th: delete the sentence for now.

Reply:
Agreed

Reply:
Agreed update done in [b293abb](https://github.com/camaraproject/QualityOnDemand/pull/17/commits/b293abb51ab2f4cc6dc0e4cea1f93cc6abd63743)